internal/archive/archive_test.go

```diff
@@ -780,3 +781,35 @@ func (s *S) testOpenArchiveArch(c *C, test realArchiveTest, arch string) {
 
 	s.checkArchitecture(c, arch, filepath.Join(extractDir, "binary"))
 }
+
+func (s *S) testProArchiveBadCreds(c *C) {
+	c.Logf("Cannot fetch Pro packages with bad credentials")
+
+	credsDir := c.MkDir()
+	restore := fakeEnv("CHISEL_AUTH_DIR", credsDir)
+	defer restore()
+
+	confFile := filepath.Join(credsDir, "credentials")
+	contents := "machine esm.ubuntu.com/fips/ubuntu/ login bearer password invalid"
+	err := os.WriteFile(confFile, []byte(contents), 0600)
+	c.Assert(err, IsNil)
+
+	options := archive.Options{
+		Label:      "ubuntu",
+		Version:    "20.04",
+		Arch:       "amd64",
+		Suites:     []string{"focal"},
+		Components: []string{"main"},
+		CacheDir:   c.MkDir(),
+		Pro:        "fips",
+		PubKeys:    []*packet.PublicKey{keyUbuntuFIPSv1.PubKey},
+	}
+
+	// The archive can be "opened" without any credentials since the dists/ path
+	// containing InRelease files, does not require any credentials.
+	testArchive, err := archive.Open(&options)
+	c.Assert(err, IsNil)
+
+	_, _, err = testArchive.Fetch("openssh-client")
+	c.Assert(err, ErrorMatches, `cannot find archive data`)
```

Comment:
I think this error message is not right, when I run Chisel locally, if I have outdated credentials I don't expect to see `cannot find archive data` with no indication that the response code was 401 Unauthorized, which clearly indicates the source of the problem.

I think we need to do two things here:
* Change the error message inside `fetch` to indicate that we were unauthorized (we don't know the reason, so we should not mention the key).
* Add a unit test which asserts that the error message is returned when return code is 401. This test does not replace `testProArchiveBadCreds`, I think we need both.

Reply:
Added 68206e5710cc9c76a0f71f24a9533dd30d08ca15.